pykern/pkcli/ci.py

```diff
@@ -44,7 +44,7 @@
 _PRINT_OK = re.compile(r"^\s*#\s*(?:pkdp|print)\(")
 
 
-def check_eof_newline():
+def check_eof_newline(github_ci_repo=None):
```

Comment:
In general, this type of parameter is better solved with config. It would be passed in python-ci downloader. However, I don't think we need to do it this way. Rather, we know this is a python ci. If it is executed in the repo root, the file setup.py will be there, for example. This can be a sentinel for specific action. In this case, what we want to limit is the top level recursions. In the walk_tree on line 125, just list the directories that are supposed to be included, e.g. package and tests. This will require an outer loop for `p` for this special case. This could be solved by creating a function that yields (generator) results for the outerloop. In the top level case, it would yield two directories. The other cases would yield a single directory.

Reply:
Can we meet about this on monday? I don't understand your description, and looking back I think I misinterpreted [your comment](https://github.com/radiasoft/pykern/issues/329#issuecomment-1352277598).

Reply:
figured it out, although I didn't see the advantage of using a generator for this case. Will send an email monday about the confusion above. 928b447ead65201a9275ac9d42f7f202d7eb5bf4